module/VuFind/src/VuFind/RecordDriver/Feature/MarcAdvancedTrait.php

```diff
@@ -120,54 +120,130 @@ public function getAccessRestrictions()
      */
     public function getAllSubjectHeadings($extended = false)
     {
-        // This is all the collected data:
-        $retval = [];
+        if (
+            isset($this->mainConfig)
+            && $this->mainConfig->get('Record') !== null
+            && $this->mainConfig->get('Record')->get('subjectHeadingsSort') === 'marc'
```

Comment:
You should be able to simplify this to `if ($this->mainConfig->Record->subjectHeadingsSort ?? '' === 'marc') {`. We don't typically use the `get()` method when accessing configurations.